src/__template__/ComponentTemplate.vue

```diff
@@ -90,6 +101,70 @@ const rootElementStyles = computed((): Record<string, string> => ({
   lineHeight: '0',
   width: iconSize.value,
 }))
+
+/**
+ * Prefix all ids within the SVG element and update all corresponding references to these ids. This is useful to avoid id conflicts when utilizing multiple instances of the same SVG in the DOM.
+ *
+ * @param {SVGElement} svgElement - The SVG element whose potential IDs need to be prefixed.
+ */
+const prefixSvgIds = (svgElement: SVGElement): void => {
+  if (!svgElement) {
+    return
+  }
+
+  const defsElement = svgElement.querySelector('defs')
+
+  // Prepare a map to store the original and new IDs for quick lookups
+  const idMap: Record<string, string> = {}
+
+  if (defsElement) {
+    // Prefix all direct children of <defs> that have an `id`
+    defsElement.querySelectorAll('[id]').forEach((element) => {
+      const originalId = element.getAttribute('id')
+      const newId = `${Math.random().toString(36).substring(2, 12)}-${originalId}`
+
+      // Map old id to new id
+      idMap[originalId!] = newId
+
+      // Update the element's id with the new prefixed id
+      element.setAttribute('id', newId)
+    })
+  }
+
+  // Update all references to these prefixed ids in attributes like `url(#id)`, `href`, `xlink:href`
+  const referencingAttributes = ['fill', 'stroke', 'filter', 'mask', 'clip-path', 'xlink:href', 'href']
+
+  // Function to update references to ids in attributes
+  const updateReferences = (element: Element): void => {
+    referencingAttributes.forEach((attr) => {
+      const attrValue = element.getAttribute(attr)
+      if (attrValue) {
+        // Match any url(#id) or href="#id" references
+        const updatedValue = attrValue.replace(/url\(#([^)]+)\)/g, (match, id) => {
```

Comment:
Since we'd want to replace the old ID everywhere, won't it be easier to just do a string replace for the old ID with the new ID, for the entire SVG?
Something like SVG → svg_string → `svg_string.replace(originalId, idMap[originalId])` ?

Reply:
I could probably just broaden this to any string that matches the id and starts with `#`

Reply:
That would also resolve https://github.com/Kong/icons/pull/391#discussion_r1772651174 since we won't need to update the new ID for speific attributes.

Reply:
Yep; just need to refactor to try and solve for SSR

Reply:
Updated